### DISCUSSION TITLE
There is a problem with the size of the packaged program file

### DISCUSSION THREAD

**Discussion by pychans** [flet-dev/flet#6187] [General]
The latest version of packaged program files is more than twice as large as before。
Windows10, Python 3.8.10

V0.19.0: `flet pack main.py`: 31,954 KB
V0.19.0: `pyinstaller -F -w main.py`: 30,651 KB
V0.22.0: `flet pack main.py`: 76,558 KB
V0.22.0: `pyinstaller -F -w main.py`:  51,296 KB
V0.23.2: flet pack main.py: 77,353 KB
V0.23.2: pyinstaller -F -w main.py: 52.127 KB
code: 
```python
import flet as ft

def main(page: ft.Page):
    page.title = "ToDo App"
    page.horizontal_alignment = ft.CrossAxisAlignment.CENTER
    page.update()

    page.add(ft.Text(value='test'))

ft.app(target=main)
```
requirements.txt：
flet
pyinstaller

**Comment by BrentHuang** [comment]
it‘s too big. Is there any way to reduce it?

**Comment by pychans** [comment]
The new version (0.22.1) has the same issue.

**Comment by pychans** [comment]
The new version (0.23.0) has the same issue.

**Comment by BrentHuang** [comment]
desktop, android apk, web,   after build, all too big size.

**Comment by BrentHuang** [comment]
Another problem is the startup speed. After build, the startup speed is slow.
 
I am using flet to make a web project and a desktop project. Customers cannot accept such a large size and such a slow startup speed. These two problems greatly affect the large-scale commercial use of Flet.

**Comment by pychans** [comment]
The new version (0.24.1) has the same issue. It looks like something was packed twice.

**Comment by pychans** [comment]
The new version (0.25.2) has the same issue, and the packaged file is even larger.
V0.25.2: flet pack main.py: 80.2 MB
V0.25.2: pyinstaller -F -w main.py: 55.4 MB

main.py：
```python
import flet as ft

def main(page: ft.Page):
    page.add(ft.Text("Hello, world!"))

ft.app(target=main)
```

**Comment by FeodorFitsner** [comment]
I guess `pyinstaller` command does not bundle Flet client executable while `flet pack` does.
Run produced app and while it's running see what's unpacked into its temporary directory which should start with `$TMPDIR/_MEI`.

**Comment by Abbos20072005** [comment]
Hello everyone, are there any solutions for these two issues with startup speed and the app size?

**Comment by josStorer** [comment]
@Abbos20072005 The only solution I can think of at the moment is to continue using v0.19.0. I am still using v0.19.0 in various utility applications for my personal projects. I believe it is the best version in terms of the trade-off between size, startup speed, and functionality. When packaged with Nuitka and some unnecessary files removed, the typical size of the compressed package on windows is about 11MB, and there is no need to install any additional dependencies. It can start almost without any delay. All subsequent versions have significantly increased in size and become slow to start.

Regarding the startup speed issue, the latest version of Flet suggests using a startup screen to mitigate it. However, I believe it is still possible to restore the original fast startup (versions v0.19.0 and earlier), but considering that the project architecture of Flet has undergone significant changes since then, I think it would require an enormous amount of work to achieve this.

---

Here is a packaged example program [main.dist.zip](https://github.com/user-attachments/files/20368228/main.dist.zip) (8.65MB) that you can run in a sandbox environment. It's packed using python3.8 + nuitka2.7.3.

`nuitka .\main.py --windows-console-mode=disable --msvc=latest --standalone`.

Note that this packing method will download the flet-windows.zip runtime environment from GitHub during the first run to share the runtime environment with other programs and reduce the size of the program. You can proactively download the flet-windows.zip of v0.19.0 and package the environment together, which will increase the size of the compressed package to 17.7MB.
